src/hooks/utils.ts

```diff
@@ -0,0 +1,26 @@
+//@ts-ignore
+export const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+type AddParamsProps = {
+  paramsArray: SingleParam[];
+  endpoint: string;
+};
+
+export type SingleParam = Record<string, string>;
+
+export const addParams = ({ paramsArray, endpoint }: AddParamsProps) => {
```

Comment:
Tbh, I don't know why you need any helper at all, it's very easy to use `URLSearchParams` directly:
```
const queryParams = { foo: 'bar' }
const search = `?${new URLSearchParams(queryParams)}`; // `?foo=bar`
```